service/storage_proxy.cc

```diff
@@ -423,14 +423,14 @@ class storage_proxy::remote {
 
         try {
             co_await coroutine::parallel_for_each(all_endpoints, [&] (auto ep) {
-                return send_truncate(netw::messaging_service::msg_addr{ep, 0}, timeout, keyspace, cfname);
+                return send_truncate(netw::messaging_service::msg_addr{ep, 0}, timeout, keyspace, cfname, frozen_topology_guard{});
             });
-           } catch (rpc::timeout_error& e) {
-               slogger.trace("Truncation of {} timed out: {}", cfname, e.what());
-               throw;
-           } catch (...) {
-               throw;
-           }
+        } catch (rpc::timeout_error& e) {
+            slogger.trace("Truncation of {} timed out: {}", cfname, e.what());
+            throw;
+        } catch (...) {
+            throw;
```

Comment:
This "default" catch block is really superfluous.
If the exception isnxt caught, it is thrown anyhow, so why do we need this?

Reply:
That default catch/throw block was introduced here: d0864be20f0e63f89da95914e12092ac3bc206dd and through the years, has survived a few changes of this function.
It might have had some purpose with the initial implementation (maybe some trickery with `handle_exception()`?), but I agree it does not seem to make sense now.
@elcallio Do you remember why we had this empty default catch/re-throw?

Reply:
I have removed the empty catch/re-throw block